opus/application/static_media/js/browse.js

```diff
@@ -561,51 +557,45 @@ var o_browse = {
         o_browse.loadData(opus.prefs.view, 1);
     },
 
-    // check if we need infiniteScroll to load next page when there is no more prefected data
-    loadNextPageIfNeeded: function(opusId) {
+    loadPageIfNeeded: function(direction, opusId) {
         // opusId will be empty at the end of the observations, so just return out.
         if (opusId === "") {
             return;
         }
         let startObsLabel = o_browse.getStartObsLabel();
         let tab = opus.getViewTab();
         let contentsView = o_browse.getScrollContainerClass();
+        o_browse.currentOpusId = opusId;
+
         let maxObs = (opus.prefs.view === "cart" ? o_cart.cartCount : opus.resultCount);
+        let obsNum = $(`${tab} .thumbnail-container[data-id=${opusId}]`).data("obs");
 
-        let obsNum = $(`${tab} .thumbnail-container[data-id=${opusId}]`).data("obs") + 1;
-        if (obsNum <= maxObs) {
-            let nextElem = $(`${tab} .thumbnail-container[data-obs=${obsNum}]`);
-            if (nextElem.length === 0) {
-                // disable keydown on modal when it's loading
-                // this will make sure we have correct html elements displayed for prev observation
-                $("#galleryViewContents").addClass("op-disabled");
-                opus.prefs[startObsLabel] = obsNum;
-                $(`${tab} ${contentsView}`).infiniteScroll("loadNextPage");
-            }
-        }
-    },
+        let checkView = (direction === "next" ? obsNum <= maxObs : obsNum > 0);
 
-    loadPrevPageIfNeeded: function(opusId) {
-        // if opusId is ever empty, return out.
-        if (opusId === "") {
-            return;
-        }
-        let startObsLabel = o_browse.getStartObsLabel();
-        let tab = opus.getViewTab();
-        let contentsView = o_browse.getScrollContainerClass();
-        o_browse.currentOpusId = opusId;
-        // decrement obsNum to see if there is a previous one to retrieve
-        let obsNum = $(`${tab} .thumbnail-container[data-id=${opusId}]`).data("obs") - 1;
-        if (obsNum > 0) {
-            let prevElem = $(`${tab} .thumbnail-container[data-obs=${obsNum}]`);
-            // if it's not there go retrieve it...
-            if (prevElem.length === 0) {
+        if (checkView) {
```

Comment:
Maybe I'm just too tired, but I find this very confusing. We just got obsNum by looking it up in the thumbnail-container by opusid. So we know it's there, right? Now we're immediately looking it up by obsNum and checking to see whether it's there or not. But it has to be there...  If it's not there, we do this weird thing where we set the startobs slug back an entire browser page. Why would we ever do that? If I'm right about the first point, this code is never called anyway.